OptimizelySDKCore/OptimizelySDKCore/Optimizely.m

```diff
@@ -314,7 +314,7 @@ - (void)track:(NSString *)eventKey
         OPTLYExperiment *experiment = [self.config getExperimentForId:eventDecision[OPTLYEventParameterKeysDecisionExperimentId]];
         OPTLYVariation *variation = [experiment getVariationForVariationId:eventDecision[OPTLYEventParameterKeysDecisionVariationId]];
         if (experiment != nil && variation != nil) {
-            experimentVariationMapping[experiment] = variation;
+            experimentVariationMapping[experiment.experimentId] = variation;
```

Comment:
@kellyroach-optimizely should we store the variation id to be consistent? Also, can you confirm that Android is doing something similar?

Reply:
REGARDING: "should we store the variation id to be consistent?".

I'd say as general principle, unless data structure will be serialized for network or persistence, it's preferable to access objects directly as themselves than less efficiently via id's.  That excuses variation values continuing to be stored as the OPTLYVariation objects themselves instead of their id's.  A problem with not doing similarly with the dictionary keys which were OPTLYExperiment's is the OPTLYExperiment's weren't "hashable" and changing the mismatched forced cast in the AppDelegate.swift code from Dictionary<String,AnyObject> to anything like Dictionary<Experiment,Variation> or Dictionary<AnyObject,AnyObject> won't compile.  That leaves us Dictionary<String,String>, Dictionary<String,Variation>, or Dictionary<String,AnyObject> to choose from.  We've ruled out Dictionary<String,String> by the preceding sentences.  We know all the values will be OPTLYVariation's so the Dictionary<String,Variation> cast is preferable as it leads to more succinct Swift code -- no follow up cast to Variation is required.

Reply:
REGARDING: "can you confirm that Android is doing something similar?"
* Case insensitive search whole project android-sdk for
    experimentVariationMapping --> NO HITS
    OptimizelyExperimentActivated --> NO HITS
* Case sensitive search android-sdk "test-app" directory for "Map<"
and examine each of the "4 occurrences".
    Map<String, Object> userProfileMap = userProfileService.lookup("test_user");
    ...
    public Map<String,String> getAttributes() {
        Map<String,String> attributes = new HashMap<>();
None of these hits resemble the objective-c-sdk dictionary in
play in this P.R. #186 .  I'm concluding Android SDK doesn't
have any similar code.

Reply:
@kellyroach-optimizely Look in the java-sdk.  ProjectConfig and Environment are defined at the java core level